abi/abi.go

```diff
@@ -0,0 +1,33 @@
+package abi
+
```

Comment:
nit: I think it would be better to put `abi` in `encoding` dir, but if there's any specific reason to move it out, let me know.

(BTW, did you up the version of `go-algorand`? It seems you are still writing `MakeTupleType` again.)

Reply:
I see, that makes very much sense.

Reply:
To be clear, I am suggesting that `interactions.go` should move into the `abi` package @algoidurovic